### PR TITLE
Allow integration tests longer timeouts

### PR DESCRIPTION
**Pull request by flomotlik**
## What did you implement:

Sometimes tests fail because integration tests have too limited timeouts set, setting timeouts higher for those integrationt tests.

## Todos:

- [x] Write tests
- [ ] Write documentation
- [x] Fix linting errors
- [x] Make sure code coverage hasn't dropped
- [x] Provide verification config/commands/resources
- [x] Leave a comment that this is ready for review once you've finished the implementation

### CODE REVIEW

File: tests/classes/CLI.js [hunk 1]
```diff
@@ -291,7 +291,8 @@ describe('CLI', () => {
       process.chdir(this.cwd);
     });
 
-    it('prints general --help to stdout', (done) => {
+    it('prints general --help to stdout', function (done) {
+      this.timeout(10000);
```
Comment (dougmoscrop) [inline tests/classes/CLI.js]: The `before` and `after` have to be changed from `() =>` to `function (done)` to match this, since it's using `this`.

**Comment by dougmoscrop** [comment]
The `before` and `after` have to be changed from `() =>` to `function (done)` to match, since the code uses `this` to store the exec command.

**Comment by eahefnawy** [comment]
yep also adding to @dougmoscrop ... you can set `this.timeout(0);` to zero to disable timeout all together if that's what you're trying to do here.

**Comment by dougmoscrop** [comment]
mocha supports a `mocha.opts` file within a directory that houses the test where you can specify various opts. Maybe it would be cleaner to revert a bit back to your regular eslint style, and just deal with timeouts that way. (so move integration_test.js to /integration/...) and add a mocha.opts file there that specifies a timeout for all integration tests of 10s or whatever?

**Comment by flomotlik** [comment]
@eahefnawy @dougmoscrop yup we need to revisit this at some point to make this nicer. Mocha also has the this.slow option to mark specific tests as slow and set a different timeout. I'm not a fan of setting the timeout to 0 as this means tests never really fail and run into a timeout. Imho we should at some point rewrite things to use slow throughout the system and then set a slow timeout, but set the normal timeout lower than 2 seconds because tests really shouldn't take 2 seconds and if they do we have to evaluate why they take so long.

**Pull request merged**